nextroute/common/location.go

```diff
@@ -44,46 +50,60 @@ func (l Locations) Unique() Locations {
 
 // Centroid returns the centroid of the locations. If locations is empty, the
 // centroid will be (0, 0).
-func (l Locations) Centroid() Location {
+func (l Locations) Centroid() (Location, error) {
```

Comment:
Should we return an invalid location if locations is empty, centroid of nothing is not in the middle of the ocean (0,0)

Reply:
also, can the Locations have an invalidLocation among them? should this be considered when calculating the centroid? Meaning: only add up lat and lon for locations that are valid locations?